subsys: ipc: icmsg: Align to NO MULTITHREADING

jaz1-nordic opened the pull request:
Adapting icmsg to work without the MULTITHREADING functionality.
Dependencies for kernel work_queue, mutexes and other functions related to running multi-threaded applications have been 'ifdefed'.

masz-nordic commented on samples/subsys/ipc/ipc_service/icmsg/remote/src/main.c:
```diff
@@ -53,19 +76,33 @@ static int send_for_time(struct ipc_ept *ep, const int64_t sending_time_ms)
+#if defined(CONFIG_MULTITHREADING)
 	LOG_INF("Perform sends for %lld [ms]", sending_time_ms);
+#else
+	printk("Perform sends for %lld [ms]\r\n", sending_time_ms);
```
How about a macro that resolves into LOG or printk+\r\n depending on multithreading?

jaz1-nordic committed:
ipc: icmsg: Align to NO MULTITHREADING

Adapting icmsg to work without the MULTITHREADING functionality.
Dependencies for kernel work_queue, mutexes and other functions
related to running multithreaded applications have been 'ifdefed'.

Signed-off-by: Jakub Zymelka <jakub.zymelka@nordicsemi.no>

jaz1-nordic committed:
samples: mbox: Align to NO MULTITHREADING

Removed k_sleep dependencies in non-multithreaded runs.

Signed-off-by: Jakub Zymelka <jakub.zymelka@nordicsemi.no>

nordic-krch commented on samples/subsys/ipc/ipc_service/icmsg/common/common.h:
```diff
@@ -17,4 +17,12 @@ struct data_packet {
 	unsigned char data[DATA_SIZE];
 };
 
+#if defined(CONFIG_MULTITHREADING)
+#define INF(...) LOG_INF(__VA_ARGS__);
+#define ERR(...) LOG_ERR(__VA_ARGS__);
+#else
+#define INF(...) printk(__VA_ARGS__); printk("\r\n");
+#define ERR(...) printk(__VA_ARGS__); printk("\r\n");
+#endif
```
we should not need this. this mapping is replicating `CONFIG_LOG_MODE_MINIMAL` behavior. Logging should work in no multi-threading (deferred mode, immediate and minimal).

jaz1-nordic committed:
samples: subsys: ipc: icmsg: Align to NO MULTITHREADING

Removed k_sleep dependencies and added 'printk' instead of LOG
in non-multithreaded runs.

Signed-off-by: Jakub Zymelka <jakub.zymelka@nordicsemi.no>

masz-nordic commented on samples/subsys/ipc/ipc_service/icmsg/remote/src/main.c:
```diff
@@ -61,11 +79,17 @@ static int send_for_time(struct ipc_ept *ep, const int64_t sending_time_ms)
 		ret = ipc_service_send(ep, &msg, mlen);
 		if (ret == -ENOMEM) {
 			/* No space in the buffer. Retry. */
+			ret = 0;
```
Is this necessary?

fabiobaltieri commented:
@masz-nordic IPC is (supposedly) maintained, please don't just unassiign PRs, at least try to reach out to the maintainers first.

gmarull commented on samples/drivers/mbox/src/main.c:
```diff
@@ -47,7 +47,11 @@ int main(void)
 	printk("Maximum TX channels: %d\n", mbox_max_channels_get_dt(&tx_channel));
 
 	while (1) {
+#if defined(CONFIG_MULTITHREADING)
 		k_sleep(K_MSEC(2000));
+#else
+		k_busy_wait(2000000);
+#endif
```
same
